<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>862</width>
    <height>667</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>OSL Dataset Visualizer</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QSplitter" name="mainSplitter">
      <property name="orientation">
       <enum>Qt::Orientation::Horizontal</enum>
      </property>
      <widget class="QWidget" name="leftPanel">
       <layout class="QVBoxLayout" name="leftLayout">
        <item>
         <widget class="QPushButton" name="loadButton">
          <property name="text">
           <string>Load OSL JSON</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="saveButton">
          <property name="text">
           <string>Save OSL JSON</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="gamesLabel">
          <property name="text">
           <string>Games</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QListView" name="videoListView"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="centerPanel">
       <layout class="QVBoxLayout" name="centerLayout">
        <item>
         <widget class="QWidget" name="videoWidgetPlaceholder" native="true">
          <property name="minimumSize">
           <size>
            <width>320</width>
            <height>240</height>
           </size>
          </property>
          <property name="styleSheet">
           <string>background: #111;</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="sliderLayout">
          <item>
           <widget class="QSlider" name="slider">
            <property name="maximum">
             <number>1000</number>
            </property>
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="timeLabel">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>00:00 / 00:00</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
        <layout class="QHBoxLayout" name="videoControlsLayout">
        <item><widget class="QPushButton" name="back5sButton"><property name="text"><string>&lt;&lt; 5s</string></property></widget></item>
        <item><widget class="QPushButton" name="back1sButton"><property name="text"><string>&lt; 1s</string></property></widget></item>
        <item><widget class="QPushButton" name="backFrameButton"><property name="text"><string>&lt;|</string></property></widget></item>
        <item><widget class="QPushButton" name="playButton"><property name="text"><string>Play</string></property></widget></item>
        <item><widget class="QPushButton" name="forwardFrameButton"><property name="text"><string>|&gt;</string></property></widget></item>
        <item><widget class="QPushButton" name="forward1sButton"><property name="text"><string>1s &gt;</string></property></widget></item>
        <item><widget class="QPushButton" name="forward5sButton"><property name="text"><string>5s &gt;&gt;</string></property></widget></item>
        </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="rightPanel">
       <layout class="QVBoxLayout" name="rightLayout">
        <item>
         <widget class="QLabel" name="annotationsLabel">
          <property name="text">
           <string>Annotations</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QListView" name="annotationListView"/>
        </item>
        <item>
         <widget class="QLabel" name="editLabel">
          <property name="text">
           <string>Edit Annotation</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QFormLayout" name="formLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="labelLabel">
            <property name="text">
             <string>Label</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="labelComboBox"/>
          </item>
          <item row="1" column="1">
  <widget class="QLabel" name="annotationTimeLabel"/>
</item>
<item row="2" column="1">
  <widget class="QPushButton" name="setTimeToVideoButton">
    <property name="text">
      <string>Set to Current Video Time</string>
    </property>
  </widget>
</item>


         </layout>
        </item>
        <item>
         <widget class="QLabel" name="metadataLabel">
          <property name="text">
           <string>Metadata</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTextEdit" name="metadataTextEdit">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="navButtonsLayout">
          <item>
           <widget class="QPushButton" name="prevButton">
            <property name="text">
             <string>Previous</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="nextButton">
            <property name="text">
             <string>Next</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
 <widget class="QPushButton" name="addAnnotationButton">
  <property name="text">
   <string>Add Annotation at Current Time</string>
  </property>
 </widget>
</item>
<item>
 <widget class="QPushButton" name="removeAnnotationButton">
  <property name="text">
   <string>Remove Selected Annotation</string>
  </property>
 </widget>
</item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>862</width>
     <height>37</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionLoad_OSL_Json"/>
    <addaction name="actionSave_OSL_JSON"/>
    <addaction name="actionOpen_Settings"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionLoad_OSL_Json">
   <property name="text">
    <string>Load OSL JSON</string>
   </property>
  </action>
  <action name="actionSave_OSL_JSON">
   <property name="text">
    <string>Save OSL JSON</string>
   </property>
  </action>
  <action name="actionOpen_Settings">
   <property name="text">
    <string>Open Settings</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
